src/prefect/blocks/core.py

```diff
@@ -790,6 +800,33 @@ async def _get_block_document(
 
         return block_document, block_document_name
 
+    @classmethod
+    @sync_compatible
+    @inject_client
+    async def _get_block_document_by_id(
+        cls,
+        block_document_id: Union[str, uuid.UUID],
+        client: Optional["PrefectClient"] = None,
+    ):
+        if isinstance(block_document_id, str):
+            try:
+                block_document_id = UUID(block_document_id)
+            except ValueError as e:
+                raise ValueError(
+                    f"Block document ID {block_document_id!r} is not a valid UUID"
+                ) from e
```

Comment:
```suggestion
                )
```
no need to add more to the traceback here